base/ltoutput.dtx

```diff
@@ -2199,10 +2199,12 @@
 % Reset these here (previously reset separately for head and foot)
 % \changes{v1.2b}{1997/04/14}
 %     {Move \cs{label} and \cs{index} (from patch file)}
+% \changes{v1.4l}{2025/01/21}{Support extended syntax
+%            for \cs{label}, \cs{index}, and \cs{glossary} (311)}
 %    \begin{macrocode}
-  \let\label\@gobble
-  \let\index\@gobble
-  \let\glossary\@gobble
+  \let\label\om@command@gobbled
+  \let\index\som@command@gobbled
+  \let\glossary\om@command@gobbled
```

Comment:
As the same set of `\let`s are used in multiple places, maybe a new wrapper could be introduced. It will be helpful in other places where `\label`, etc., need to be disabled.

Reply:
what other placxes would you suggest could be needed? I can see only the two and I'm not sure it is worth changing that to become a wrapper right now. Eventually, I see that these places become hooks or sockets and then it might make sense.

Reply:
Direct examples that I can think of are from packages.

For example in `hyperref`, when generating pdf string from sectioning titles and in theorem packages like `thmtools` and `keytheorems` which provide a way to restate theorems.

So it seems a public interface is needed.

Reply:
yes eventually most likely, but at this stage we think it is too early